clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -72,13 +76,24 @@
     private static final String ISR_KEY_NAME = "isr";
 
     private final Collection<Node> brokers;
+    private final Node controller;
     private final List<TopicMetadata> topicMetadata;
 
+    /**
+     * Constructor for the latest version
+     */
+    public MetadataResponse(List<Node> brokers, int controllerId, List<TopicMetadata> topicMetadata) {
```

Comment:
@ijuma I had maintained the old constructor for compatibility in an older version of the patch. But in previous reviews I was told I did not need to maintain it: https://github.com/apache/kafka/pull/1095#discussion_r58058004